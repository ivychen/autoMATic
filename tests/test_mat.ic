/* Create a matrix */
// Assign to matrix literal
int main() {
    bool is_square;
		int matrix[3][3] M;
		int matrix[1][3] N;
		is_square = true;

		if (is_square == true) {
			M = [[1,2,3];[3,2,1];[3,3,3]];
			return M;
		} else {
			// 1D matrix
			N = [[1,2,3]];
			return N;
		}
}
