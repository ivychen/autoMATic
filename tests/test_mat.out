
int mat(is_square)
{
matrix M;
if (is_square == true)
{
M = [[1,2,3];[3,2,1];[3,3,3]];
}
else
{
M = [[1,2,3]];
}
return M;
}
